clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -80,19 +80,30 @@ void CodeGenTypes::addRecordTypeName(const RecordDecl *RD,
                 OS << "i" << TTy->getIntegerBitWidth();
                 break;
               }
-            } else if (TTy->isBFloatTy())
+            } else if (TTy->isHalfTy()) {
+              OS << "half";
```

Comment:
I'm not opposed to the approach since integer types are printed in the same way in the code above, but could you please elaborate what is the problem with `print` function?